Source/WebCore/editing/ReplaceSelectionCommand.cpp

```diff
@@ -1808,6 +1808,21 @@ ReplacementFragment* ReplaceSelectionCommand::ensureReplacementFragment()
     return m_replacementFragment.get();
 }
 
+static bool fullySelectsEnclosingLink(const VisibleSelection& selection)
+{
+    auto start = selection.start();
+    auto end = selection.end();
+    RefPtr ancestor = commonInclusiveAncestor(start, end);
+    if (!ancestor)
+        return false;
+
+    RefPtr link = ancestor->enclosingLinkEventParentOrSelf();
+    if (!link)
+        return false;
+
+    return positionBeforeNode(link.get()).downstream().equals(start) && positionAfterNode(link.get()).upstream().equals(end);
```

Comment:
What about cases where we've selected slightly before the start or slightly after the end?

Reply:
Ah, so in the case where the selection starts slightly before the link or ends slightly after the link, we should still be avoiding the trivial replacement codepath since the start/end container nodes won't match (which means the entire link should still be replaced with text). I'll add a couple more cases to the layout test to verify this behavior.